src/typecheck.rs

```diff
@@ -747,9 +752,6 @@ impl EGraph {
                     let value = if let Some(out) = function.nodes.get(values) {
                         out.value
                     } else if make_defaults {
-                        if function.merge.on_merge.is_some() {
```

Comment:
`make_defaults` implies that this isn't a datatype. Either there is a merge function, or you get an exception because you tried to merge two primitives.